x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -182,25 +183,28 @@ const ColorPicker = ({
 
   const handleColor: EuiColorPickerProps['onChange'] = (text, output) => {
     setColor(text);
-
     if (output.isValid || text === '') {
-      const newYConfigs = [...(layer.yConfig || [])];
-      const existingIndex = newYConfigs.findIndex((yConfig) => yConfig.forAccessor === accessor);
-      if (existingIndex !== -1) {
-        if (text === '') {
-          delete newYConfigs[existingIndex].color;
-        } else {
-          newYConfigs[existingIndex].color = output.hex;
-        }
+      updateColorInState(text, output);
+    }
+  };
+
+  const updateColorInState: EuiColorPickerProps['onChange'] = debounce((text, output) => {
```

Comment:
I think this needs to get memoized as well, otherwise there will be a new debounce instance on every render